scripts/paper/generate_stwo_phase30_source_bound_manifest_reuse_figure.py

```
@@ -0,0 +1,455 @@
+    return grouped
+
+
+def timing_metadata(rows: list[dict[str, str]], *, fallback_runs: int) -> tuple[bool, int]:
+    first = rows[0]
+    mode = first.get("timing_mode", "").strip()
+    runs_raw = first.get("timing_runs", "").strip()
+    if not mode:
+        return False, fallback_runs
+    for row in rows[1:]:
+        if row.get("timing_mode", "").strip() != mode or row.get("timing_runs", "").strip() != runs_raw:
+            raise SystemExit("inconsistent timing metadata across phase30 manifest benchmark rows")
```

Ruff (0.15.11)

[warning] Avoid specifying long messages outside the exception class

(TRY003)